### PR TITLE
Refactor getPosition() -> getModel() and add docs to MatrixStack.

### LINKED ISSUE #216: getModel -> getPosition rename was a mistake
So, I don't have the class in mind but y'all we see what I mean with this code excerpt:

```java
MatrixStack matrices;

/* code */

matrices.peek().getModel()
```

`getModel` was renamed to `getPosition` during the big rendering refactor, we imported lot of mojmap names woooo so we finally have stuff be accurate.

Except this name.
No it's not a position, it's a model matrix.
"wdym it's not a position?"

It contains **both** a translation and a rotation of the model.
It's not just a position it's more.

If we want to be the most accurate possible it should be `getModelMatrix` but that's a tad long. And given where it's used in the code, it's a bit useless since, well, we already usually know that it's a matrix.

The question though is, should this rename happen once again?
This would fuel the possible complaints of QM being unstable, but it's also very bothering to me to keep this name.
So please voice your opinion below!

### PR DESCRIPTION
**Pull request by LambdAurora**
Closes #216. You can find the reasoning for the refactor there.

Also went on and documented the class because that's always nice!

### CODE REVIEW

**Review by NoComment1105 — APPROVED** [review]

File: mappings/net/minecraft/client/util/math/MatrixStack.mapping [hunk 1]
```diff
@@ -1,26 +1,54 @@
 CLASS net/minecraft/unmapped/C_cnszsxvd net/minecraft/client/util/math/MatrixStack
+	COMMENT Represents a stack of matrices.
+	COMMENT <p>
+	COMMENT You can {@link MatrixStack#push() push} and {@link MatrixStack#pop() pop} matrices onto the stack.
+	COMMENT <p>
+	COMMENT All other operations happen on the last matrix pushed onto the stack.
+	COMMENT Those operations can be: translation, scaling, rotation, multiplication with other matrices, etc.
+	COMMENT Popping a matrix will have the effect to revert all the changes made to it.
```
Comment (Leo40Git) [inline mappings/net/minecraft/client/util/math/MatrixStack.mapping]:
```suggestion
	COMMENT Popping a matrix will have the effect of reverting all changes made to it.
```

File: mappings/net/minecraft/client/util/math/MatrixStack.mapping [hunk 2]
```diff
@@ -1,26 +1,54 @@
...
+		FIELD f_qmkzpkwl modelMatrix Lnet/minecraft/unmapped/C_gxvddjib;
+		METHOD <init> (Lnet/minecraft/unmapped/C_gxvddjib;Lnet/minecraft/unmapped/C_crutkuff;)V
+			ARG 1 modelMatrix
+			ARG 2 normalMatrix
 		METHOD m_hmmszgvl getNormal ()Lnet/minecraft/unmapped/C_crutkuff;
-		METHOD m_sgddbcul getPosition ()Lnet/minecraft/unmapped/C_gxvddjib;
+		METHOD m_sgddbcul getModel ()Lnet/minecraft/unmapped/C_gxvddjib;
+			COMMENT {@return the model matrix}
+			COMMENT <p>
+			COMMENT The model matrix contains position and rotation data for a model to render.
```
Comment (Leo40Git) [inline mappings/net/minecraft/client/util/math/MatrixStack.mapping]:
```suggestion
			COMMENT The model matrix contains the position and rotation data needed for a model to render.
```

File: mappings/net/minecraft/client/util/math/MatrixStack.mapping [hunk 3]
```diff
@@ -1,26 +1,54 @@
 CLASS net/minecraft/unmapped/C_cnszsxvd net/minecraft/client/util/math/MatrixStack
+	COMMENT Represents a stack of matrices.
+	COMMENT <p>
+	COMMENT You can {@link MatrixStack#push() push} and {@link MatrixStack#pop() pop} matrices onto the stack.
+	COMMENT <p>
+	COMMENT All other operations happen on the last matrix pushed onto the stack.
```
Comment (Leo40Git) [inline mappings/net/minecraft/client/util/math/MatrixStack.mapping]:
```suggestion
	COMMENT All other operations are applied to the last matrix pushed onto the stack.
```

**Review by Leo40Git — APPROVED** [review]
revert the atrocity, o foxolotl
Anyway, LGTM! Just some slight Javadoc rewords...

File: mappings/net/minecraft/client/util/math/MatrixStack.mapping [hunk 4]
```diff
@@ -1,26 +1,54 @@
 CLASS net/minecraft/unmapped/C_cnszsxvd net/minecraft/client/util/math/MatrixStack
+	COMMENT Represents a stack of matrices.
+	COMMENT <p>
+	COMMENT You can {@link MatrixStack#push() push} and {@link MatrixStack#pop() pop} matrices onto the stack.
+	COMMENT <p>
+	COMMENT All other operations are applied to the last matrix pushed onto the stack.
+	COMMENT Those operations can be: translation, scaling, rotation, multiplication with other matrices, etc.
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/util/math/MatrixStack.mapping]: Grammar nitpick, but I believe `scaling` should just be `scale` here to be consistent with the noun forms of the other operations. 

Alternatively, changing them all to verb form is slightly fewer characters: `translate, scale, rotate`... (scale is noun *and* verb form)

**Review by supersaiyansubtlety — COMMENTED** [review]
Minor nitpick

**Review by IotaBread — APPROVED** [review]

File: mappings/net/minecraft/client/util/math/MatrixStack.mapping (hunk 4 above)
Comment (LambdAurora) [reply mappings/net/minecraft/client/util/math/MatrixStack.mapping]: Should be fixed!

**Review by supersaiyansubtlety — APPROVED** [review]

**Pull request merged**